### ISSUE TITLE
Steps to trace embedded jetty logs using log4j2.

### ISSUE THREAD

**Issue by avin15cs** [eclipse/jetty.project#7648]
**Jetty version**
9.4.45
**Java version**
11.0.5
**Question**
What is the correct step to trace embedded jetty logs using log4j2? I went through jetty documentation but exact steps where not mentioned.

Thanks.

**Comment by olamy** [comment]
maybe it would be better to use slf4j on the top of log4j2.
I have just added an example here https://github.com/jetty-project/embedded-jetty-logging-examples/tree/jetty-9.4.x/slf4j-log4j2-logging

**Comment by rfox12** [comment]
You could do something like this at the top of your program...
```
Configurator.setRootLevel(Level.DEBUG);
```
or to dig into a specific area...
```
Configurator.setLevel("org.eclipse.jetty.server.session", Level.INFO);
```

**Comment by joakime** [comment]
Are you asking about Log4j's TRACE level?

**Comment by avin15cs** [comment]
@joakime , Yes!

**Comment by joakime** [comment]
Jetty 9.4.x doesn't use TRACE level logging (along with other common event levels) with it's own ancient (circa 1995) logging facade.

See: https://javadoc.io/static/org.eclipse.jetty/jetty-util/9.4.45.v20220203/org/eclipse/jetty/util/log/Logger.html

However, when mapping the Jetty 9.4.x logging facade to other implementations (implementations other than Jetty's own StdErrLog), the levels are mapped as such ...

Jetty Logger | java.util.logger | slf4j | log4j \*
---- | ---- | ---- | ---
ALL | ALL | (impl) | ALL
IGNORE | FINEST |  DEBUG | DEBUG
n/a | FINER | n/a | n/a
n/a | n/a | n/a | TRACE
DEBUG | FINE | DEBUG | DEBUG
n/a | CONFIG | n/a | n/a
INFO | INFO | INFO | INFO
WARN | WARNING | WARN | WARN
n/a | SEVERE | ERROR | ERROR
n/a | n/a | n/a | FATAL
OFF | OFF | (impl) | OFF

**\*** log4j is not used directly, but via [slf4j bridge APIs](https://www.slf4j.org/legacy.html).
Note that Jetty uses "Jetty Logger" levels only.
The levels ALL & OFF are present in the API for the column indicated, but you cannot create events at that level (eg: `Log.off("Hello")` doesn't exist).  ALL & OFF exists in those APIs to configure the appenders for the named logger, no events would flow through the system tagged at those levels.
Jetty 10.0.x has dropped the Jetty Logger facade and is using slf4j directly now.

In short, Jetty has no "TRACE" level logging events that it produces.
And none of the Jetty event levels will map to a log4j TRACE level.
The only way you'll see log4j TRACE events, is if something non-Jetty uses log4j directly to create events.
